pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -274,6 +294,47 @@ func (r *DNSPolicyReconciler) readyCondition(targetNetworkObjectectKind string,
 	return cond
 }
 
+func (r *DNSPolicyReconciler) enforcedCondition(ctx context.Context, targetNetworkObjectKind string, dnsPolicy *v1alpha1.DNSPolicy) (*metav1.Condition, error) {
+	dnsRecords, err := r.dnsHelper.getDNSRecordsForDNSPolicy(ctx, dnsPolicy)
+	if err != nil {
+		return nil, err
+	}
+
+	// If no records have been created at all, set the Enforced condition to
+	// unknown
+	if len(dnsRecords) == 0 {
+		return &metav1.Condition{
+			Type:    string(conditions.ConditionTypeEnforced),
+			Status:  metav1.ConditionUnknown,
+			Message: "No DNSRecords created",
+			Reason:  "DNSPolicyNoRecordsCreated",
+		}, nil
+	}
+
+	notReadyRecords := slice.Filter(dnsRecords, slice.Not((v1alpha1.DNSRecord).IsReady))
+	allRecordsReady := len(notReadyRecords) == 0
+
+	if allRecordsReady {
+		return &metav1.Condition{
+			Type:    string(conditions.ConditionTypeEnforced),
+			Status:  metav1.ConditionTrue,
+			Message: "All DNSRecords ready",
+			Reason:  "DNSPolicyAllRecordsReady",
+		}, nil
+	}
+
+	notReadyRecordNames := slice.Map(notReadyRecords, func(dnsRecord v1alpha1.DNSRecord) string {
+		return fmt.Sprintf("%s/%s", dnsRecord.Namespace, dnsRecord.Namespace)
+	})
+
+	return &metav1.Condition{
+		Type:    string(conditions.ConditionTypeEnforced),
+		Status:  metav1.ConditionUnknown,
+		Message: fmt.Sprintf("DNSRecords %v not Ready", notReadyRecordNames),
+		Reason:  "DNSPolicySomeRecordsNotReady",
```

Comment:
The docs https://github.com/Kuadrant/architecture/blob/main/rfcs/0004-policy-status.md#conditions would suggest this is unknown but given the status is Uknown already I'm not sure what the reason should be here. 🤔

Reply:
Looking at the document again I'm guessing the Status should be `False` and the Reason should be `Unknown`, which kinda makes sense too. I'll change it to that to be more consistent as I guess other Kuadrant policies will be following that